src/apps/metastreet/ethereum/metastreet.lending-v2.contract-position-fetcher.ts

```diff
@@ -0,0 +1,362 @@
+import { Inject } from '@nestjs/common';
+import { BigNumberish, BigNumber, constants } from 'ethers';
+import { gql } from 'graphql-request';
+import { sumBy } from 'lodash';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { gqlFetch } from '~app-toolkit/helpers/the-graph.helper';
+import { ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { drillBalance } from '~app-toolkit/helpers/drill-balance.helper';
+
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+  GetDefinitionsParams,
+  GetTokenDefinitionsParams,
+  UnderlyingTokenDefinition,
+  GetDisplayPropsParams,
+  GetTokenBalancesParams,
+  GetDataPropsParams,
+} from '~position/template/contract-position.template.types';
+import { ContractPositionBalance } from '~position/position-balance.interface';
+import { MetaType } from '~position/position.interface';
+
+import { MetastreetContractFactory, Pool, PoolLegacy } from '../contracts';
+
+export const GET_POOLS_QUERY = gql`
+  {
+    pools {
+      implementationVersionMajor
+      id
+      ticks {
+        id
+        limit
+        duration
+        rate
+        raw
+      }
+      currencyToken {
+        symbol
+      }
+      collateralToken {
+        name
+      }
+    }
+  }
+`;
+
+function getAllDepositsQuery(address: string) {
+  return gql`
+    {
+      pools {
+        deposits(where: { account: "${address}" }) {
+          shares
+          tick {
+            id
+            value
+            shares
+          }
+          redemptions {
+            shares
+          }
+        }
+      }
+    }
+  `;
+}
+
+export type GetPoolsResponse = {
+  pools: {
+    id: string;
+    implementationVersionMajor: string;
+    ticks: {
+      id: string;
+      limit: BigNumber;
+      duration: BigNumber;
+      rate: BigNumber;
+      raw: BigNumber;
+    }[];
+    currencyToken: string;
+    collateralToken: string;
+  }[];
+};
+
+export type GetAllDepositsResponse = {
+  pools: {
+    deposits: {
+      shares: BigNumber;
+      tick: {
+        id: string;
+        shares: BigNumber;
+        value: BigNumber;
+      };
+      redemptions: {
+        shares: BigNumber;
+      }[];
+    }[];
+  }[];
+};
+
+export const SUBGRAPH_URL = 'https://api.thegraph.com/subgraphs/name/metastreet-labs/metastreet-v2-beta';
```

Comment:
Querying the subgraphs is free

Reply:
It is for now, but it won't be indefinitely. TheGraph already have reached out to a bunch of protocol (us included) asking to move the the decentralized solution because hosted services won't be supported soon

Reply:
I see, we will cover the cost of queries as needed.